src/sql/workbench/parts/dashboard/widgets/properties/propertiesWidget.component.ts

```diff
@@ -111,16 +139,104 @@ export class PropertiesWidgetComponent extends DashboardWidget implements IDashb
 		this._changeRef.detectChanges();
 	}
 
-	private parseProperties() {
-		const provider = this._config.provider;
-
-		let propertyArray: Array<Property>;
-
+	private parseConfig(): void {
 		// if config exists use that, otherwise use default
 		if (this._config.widget['properties-widget'] && this._config.widget['properties-widget'].properties) {
 			const config = <PropertiesConfig>this._config.widget['properties-widget'];
-			propertyArray = config.properties;
-		} else {
+			this.parseProperties(config.properties);
+		}
+		else if (this._config.widget['properties-widget'] && this._config.widget['properties-widget'].endpoints) {
+			this._isVertical = true;
+			this.parseEndpoints();
+		}
+		else {
+			this.parseProperties(null);
+		}
+		this._loading = false;
+	}
+
+	private parseEndpoints(): void {
+		if (!this.endpoints || this.endpoints.length === 0) {
+			this.getEndpoints();
+		}
+
+		// iterate over endpoint properties and display them
+		this.properties = [];
+		for (const endpoint of this.endpoints) {
+			let propertyObject = this.getValueOrDefault<string>(endpoint, EndpointConstants.hyperlink.toString(), '--');
+			const assignProperty: DisplayProperty = {
+				displayName: endpoint.serviceName,
+				value: propertyObject,
+				isHyperlink: this.isHyperlink(endpoint.serviceName)
+			};
+
+			this.properties.push(assignProperty);
+		}
+
+		if (this._hasInit) {
+			this._changeRef.detectChanges();
+		}
+	}
+
+	private getEndpoints(): void {
+		const endpointArray = this._connection.serverInfo.options[EndpointConstants.clusterEndpoints.toString()];
+
+		// TO DO: Once the endpoints data is moved to DMV, update this logic to replace hardcoded values
+		// with the data from DMV.
+		if (endpointArray && endpointArray.length > 0) {
+			// iterate over endpoints and display them
+			this.endpoints = [];
+			for (const ep of endpointArray) {
+				const assignProperty: Endpoint = {
+					serviceName: ep.serviceName,
+					ipAddress: ep.ipAddress,
+					port: ep.port,
+					hyperlink: ep.ipAddress + ':' + ep.port
+				};
+				this.endpoints.push(assignProperty);
+			}
+
+			// add grafana and kibana endpoints from the management proxy endpoint
+			const managementProxyEp = endpointArray.find(e => e.serviceName === EndpointConstants.managementproxy.toString());
+			if (!endpointArray.find(e => e.serviceName.toLowerCase().indexOf('metrics') > -1)) {
```

Comment:
>if (!endpointArray.find(e => e.serviceName.toLowerCase().indexOf('metrics') > -1)) { [](start = 3, length = 84)

All this logic seems really really fragile - is there really no better way to determine which endpoints to display?

Reply:
right now, no (even the previous portal had it this way). But as I added, this really is temporary, and they are deprecating the table and moving it to dmv. So once they give us the dmv (fingers crossed) all this will be deleted.